src/javascript/app_2/Stores/client_store.js

```diff
@@ -1,12 +1,327 @@
-import { observable } from 'mobx';
-import Client         from '_common/base/client_base';
-import BaseStore      from './base_store';
+import { observable, computed, action, reaction } from 'mobx';
+import moment                                     from 'moment';
+import GTM                                        from '_common/base/gtm';
+import * as SocketCache                           from '_common/base/socket_cache';
+import BinarySocket                               from '_common/base/socket_base';
+import eventBus                                   from 'Services/event_bus';
+import { LocalStore, State }                      from '_common/storage';
+import BaseStore                                  from './base_store';
+import { localize }                               from '../../_common/localize';
+
+const types_map = {
+    virtual  : 'Virtual',
+    gaming   : 'Gaming',
+    financial: 'Investment',
+};
 
 export default class ClientStore extends BaseStore {
-    @observable balance;
-    @observable is_logged_in = !!Client.isLoggedIn();
-    @observable loginid      = Client.get('loginid');
-    @observable currency     = Client.get('currency') || '';
-    @observable upgrade_info = Client.getBasicUpgradeInfo();
-    @observable can_upgrade  = this.upgrade_info.can_upgrade || this.upgrade_info.can_open_multi;
+    @observable loginid;
+    @observable upgrade_info;
+    @observable accounts;
+    storage_key = 'client.accounts';
+    @observable switched    = '';
+
+    /**
+     * Borrowed from `Client_base::getBasicUpgradeInfo()`
+     * @returns {{type: *, can_upgrade: boolean, can_upgrade_to: *, can_open_multi: boolean}}
+     */
+    getBasicUpgradeInfo = () => {
```

Comment:
@easteregg @negar-binary 
isn't it better to keep the business logic for both the legacy and the new app in one place instead of duplicating them?
Whereas the logic of keeping both projects in one repository was to maintain them better and reduce the duplication.
It seems that is better to refactor the Client_base and extract logics which are common between both projects, in this way, there is just one code_base for client logic and both `Client_base` and `client_store` use it.

Reply:
makes sense, since we are going to continue keeping them in the same repo for now.

Reply:
yes, by extracting common business logic to another place as a black-box functions, both `Client_base` and `client_store` can use them just by passing required parameters to them. Also, there is just one code-base for client business logic for both of them.